components/ResourceCreation/TestResourcesDisplay.tsx

```diff
@@ -5,43 +5,101 @@ import { patientTestCaseState } from '../../state/atoms/patientTestCase';
+import { searchQueryState } from '../../state/atoms/searchQuery';
+import { useEffect, useState } from 'react';
+import { DataRequirement } from 'fhir/r4';
+
+interface DataSearchKeys {
+  type: string;
+  id: string | undefined;
+  displayString: string;
+}
 
 export default function TestResourcesDisplay() {
   const dataRequirements = useRecoilValue(dataRequirementsState);
   const [, setSelectedDataRequirement] = useRecoilState(selectedDataRequirementState);
   const valueSetMap = useRecoilValue(valueSetMapState);
   const currentPatients = useRecoilValue(patientTestCaseState);
+  const searchQuery = useRecoilValue(searchQueryState);
+  const [dataReqsToDisplay, setDataReqsToDisplay] = useState<DataRequirement[] | null>(dataRequirements);
 
-  return dataRequirements?.length && Object.keys(currentPatients).length ? (
-    <div
-      data-testid="test-resource-panel"
-      style={{
-        maxHeight: '40vh',
-        overflow: 'scroll'
-      }}
-    >
-      <Stack>
-        {dataRequirements.map((dr, i) => {
-          const key = `${dr.type}${i}`;
+  useEffect(() => {
+    /**
+     * Assembles an array of data that will be searched by FuseJS to narrow
+     * down the displayed data requirements in the test resources display.
+     * @returns array of data to use for fuzzy searching
+     */
+    const getSearchContent = (): DataSearchKeys[] | undefined => {
+      if (dataRequirements) {
+        const searchableData = dataRequirements.map(dr => {
+          const type = dr.type;
+          const id = dr.id;
           const displayString = getDataRequirementFiltersString(dr, valueSetMap);
-          return (
-            <Button
-              key={key}
-              onClick={() => {
-                setSelectedDataRequirement({ name: displayString, content: dr });
-              }}
-              fullWidth
-              variant="default"
-              style={{ height: 'auto', padding: 10, overflow: 'hidden' }}
-            >
-              <div style={{ overflow: 'hidden' }}>
-                <Title order={3}>{dr.type}</Title>
-                <Text style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>{displayString}</Text>
-              </div>
-            </Button>
-          );
-        })}
-      </Stack>
-    </div>
+          return { type, id, displayString };
+        });
+        return searchableData;
+      }
+    };
+
+    /**
+     * Uses FuseJS to do a fuzzy search on the search contents. Then, filters the
+     * data requirements so that the displayed data requirements are those that
+     * came back from the search.
+     * @returns array of data requirements returned from FuseJS search
+     */
+    const filterBySearchResults = () => {
+      const searchContents = getSearchContent();
+      if (dataRequirements && searchContents) {
+        const keys: (keyof DataSearchKeys)[] = ['type', 'id', 'displayString'];
+        const fuse = new Fuse(searchContents, { keys: keys });
```

Comment:
Thanks for looking at the different thresholds - I was talking with Elsa yesterday about how the threshold is most likely too high because it sometimes returns resource types that are not even remotely mentioned in the search. I'll play around with some threshold values and see what works best 🌡️ (whenever I try to type a thumbs up it switches it to a thermometer so I'm keeping it)